geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geohash/GeohashUtils.scala

```diff
@@ -521,12 +520,14 @@ object GeohashUtils
       }
 
       if (numBoxes==0)
-        throw new Exception("Could not satisfy constraints, resolutions " + resolutions + ", constraints " + constraints + ".")
+        throw UnsatisfiedConstraintsException(s"Could not satisfy constraints, resolutions $resolutions, constraints $constraints.")
 
       RecommendedResolution(maxBits, numBoxes)
     }
   }
 
+  case class UnsatisfiedConstraintsException(message: String) extends Exception(message)
```

Comment:
@jnh5y We don't actually need the case class for case matching anymore. So we can either get rid of this PR all together or use it as an opportunity to change this to an illegal argument exception. Let me know what you think is best.